common/zeroconf/zeroconf-qt.h

```diff
@@ -0,0 +1,86 @@
+/*
+	Aseba - an event-based framework for distributed robot control
+	Copyright (C) 2007--2016:
+		Stephane Magnenat <stephane at magnenat dot net>
+		(http://stephane.magnenat.net)
+		and other contributors, see authors.txt for details
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Lesser General Public License as published
+	by the Free Software Foundation, version 3 of the License.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Lesser General Public License for more details.
+
+	You should have received a copy of the GNU Lesser General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef ASEBA_ZEROCONF_QT
+#define ASEBA_ZEROCONF_QT
+
+#include <QSocketNotifier>
+#include <QSignalMapper>
+#include "zeroconf.h"
+
+namespace Aseba
+{
+	//! Use the QT event loop to watch the DNS Service for updates, and triggers callback
+	//! processing as necessary.
+	class QtZeroconf : public QObject, public Zeroconf
+	{
+		Q_OBJECT
+
+	public:
+		void browse();
+
+	signals:
+		void zeroconfBrowseCompleted(); //!< emitted when browsing is completed
+		void zeroconfRegisterCompleted(const Aseba::Zeroconf::Target *); //!< emitted when a register is completed
```

Comment:
This signal passes a pointer to the target, however this signal might be connected to another thread through a `Qt::QueuedConnection`. In that case, are we sure that the target pointer will still be valid at that time. I guess this pointer points to a member of `targets` in the main `Zeroconf` object.

If this is a problem, we could split the current `Zeroconf::Target` object into a parent that contains all the informations but the request itself and the pointer to the container, and a child that adds these information. Then, a copy of the parent would be passed to the callee slot.